application/src/main/java/org/togetherjava/tjbot/features/moderation/TransferQuestionCommand.java

```diff
@@ -0,0 +1,220 @@
+
+            message = message.substring(0, lastWordEnd);
+        }
+
+        return isTitleValid(message) ? message : "Untitled";
+    }
+
+    private static boolean isTitleValid(CharSequence title) {
+        String titleCompact = TITLE_COMPACT_REMOVAL_PATTERN.matcher(title).replaceAll("");
+
+        return titleCompact.length() >= TITLE_COMPACT_LENGTH_MIN
+                && titleCompact.length() <= TITLE_COMPACT_LENGTH_MAX;
+    }
+
+    private RestAction<ForumPost> createForumPost(ModalInteractionEvent event, User originalUser) {
+
+        String originalMessage = event.getValue(TRANSFER_QUESTION_INPUT_ID).getAsString();
+
+        MessageEmbed embedForPost = makeEmbedForPost(originalUser, originalMessage);
+
+        MessageCreateData forumMessage = new MessageCreateBuilder()
+            .addContent("%s has a question:".formatted(originalUser.getAsMention()))
+            .setEmbeds(embedForPost)
+            .build();
+
+        String forumTitle = event.getValue(TRANSFER_QUESTION_TITLE_ID).getAsString();
+        String transferQuestionTag = event.getValue(TRANSFER_QUESTION_TAG).getAsString();
+
+        ForumChannel questionsForum = getHelperForum(event.getJDA());
+
+        String queryTag = StringDistances.closestMatch(transferQuestionTag, defaultTags)
+            .orElse(defaultTags.get(0));
+
+        ForumTag defaultTag = getDefaultTagOr(questionsForum.getAvailableTagsByName(queryTag, true),
+                () -> questionsForum.getAvailableTagsByName(defaultTags.get(0), true).get(0));
+
+        return questionsForum.createForumPost(forumTitle, forumMessage)
+            .setTags(ForumTagSnowflake.fromId(defaultTag.getId()))
+            .map(createdPost -> new ForumPost(originalUser, createdPost.getMessage()));
+    }
+
+    private RestAction<Message> dmUser(MessageChannelUnion sourceChannel, ForumPost forumPost,
+            Guild guild) {
+
+        return forumPost.author.openPrivateChannel()
+            .flatMap(channel -> channel.sendMessage(
+                    """
+                            Hello 👋 You have asked a question on %s in the wrong channel. Not a big deal, but none of the experts who could help you are reading your question there 🙁.
+
+                            Your question has been automatically transferred to %s, please continue there. You might also want to give #welcome a quick read, thank you 👍.
+                            """
+                        .formatted(guild.getName(), forumPost.message.getJumpUrl())))
+            .onErrorFlatMap(error -> sourceChannel.sendMessage(
+                    """
+                            Hello %s 👋 You have asked a question in the wrong channel. Not a big deal, but none of the experts who could help you are reading your question there 🙁.
+
+                            Your question has been automatically transferred to %s, please continue there. You might also want to give #welcome a quick read, thank you 👍.
+                            """
```

Comment:
duplication of the text. it looks almost identical. whats the difference? the ping?

get rid of the duplication. for example by using the `Hello %s ...` text and then conditionally formatting that either with `""` or the mention.

also, if we cant make `#welcome` clickable (in this iteration), id rather remove the last sentence entirely and write:
```java
... please continue there, thank you 👍
```
(also, if u end on a smiley, dont put a dot)

Reply:
not sure how to do this one, these messages are for two different calls. Could you maybe suggest something?

Reply:
prepare both texts beforehand with variables:

```java
String textTemplate = """
  Hello %s 👋 You have asked a question on %s in the wrong channel.
  ...
  ...
""";

String dmText = textTemplate.formatted("", guild.getName(), forumPost.message.getJumpUrl());
String channelText = textTemplate.formatted(forumPost.author.getAsMention(), "", forumPost.message.getJumpUrl());
```
And then later:
```java
.flatMap(channel -> channel.sendMessage(channelText))
.onErrorFlatMap(error -> sourceChannel.sendMessage(dmText))
```

Reply:
ah oki this one makes sense, i thought you want to conditionally do both the calls in one line or something.